### PR TITLE
Add trompeloeil C++ mocking framework

### PR DESCRIPTION
**Pull request by rollbear**
Appears to work when tested locally

### CODE REVIEW

File: etc/config/c++.amazon.properties [hunk 1]
```diff
@@ -1132,6 +1132,35 @@ libs.tomlplusplus.versions.133.version=1.3.3
 libs.tomlplusplus.versions.133.path=/opt/compiler-explorer/libs/tomlplusplus/v1.3.3/include
 libs.tomlplusplus.versions.124.version=1.2.4
 libs.tomlplusplus.versions.124.path=/opt/compiler-explorer/libs/tomlplusplus/v1.2.4/include
+libs.trompeloeil.name=trompeloeil
+libs.trompeloeil.versions=40:39:38:37:36:35:34:33:32:31:30:29:28
+libs.trompeloeil.url=https://github.com/rollbear/trompeloeil
+libs.trompeloeil.versions.40.path=/opt/compiler-explorer/libs/trompeloeil/v40/include
+libs.trompeloeil.versions.40.version=v40
```
Comment (mattgodbolt) [inline etc/config/c++.amazon.properties:1139]: The `v40` is the name that will appear in the dropdown - is this the "official" version you'd like users to see, or would you prefer it to appear as "40" ?

Comment (rollbear) [reply etc/config/c++.amazon.properties:1139]: v40 is the "official" name, if there is such a thing. It's the name referred to in the ChangeLog.

Comment (mattgodbolt) [reply etc/config/c++.amazon.properties:1139]: Perfect! thanks!

**Review by mattgodbolt — APPROVED** [review]

**Comment by mattgodbolt** [comment]
This is now live. the `v` prefix doesn't work that well with our semantic version ordering; so the most recent is last in the list. We can work on that if it's an issue. Thanks for the PR!!

**Comment by rollbear** [comment]
:+1:

**Pull request merged**